### PR TITLE
make sure vcr handler tests run

### PR DESCRIPTION
**Pull request by lbouchard-adgear**
Also got rid of dummyprotocol package along the way. 

I replaced it by a small mock protocol in handler_test.go.  It was not used elsewhere.

### CODE REVIEW

File: barker/vcr/testutil/testutil.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package testutil
+
+import (
+	proto "github.com/golang/protobuf/proto"
+)
+
+func Equal(expected proto.Message, actual proto.Message) bool {
```
Comment (Medulli) [inline barker/vcr/testutil/testutil.go:7]: Nice stuff ! I don't think that's specific to `vcr` though, I'd put that under `/tests/`. That's very nitpicky though 😄

Comment (lbouchard-adgear) [reply barker/vcr/testutil/testutil.go:7]: oh you're right.
I have no idea of what is a good standard location in go for test utils ... Will put it under /tests for now. If someone has an objection let me know.

**Comment by juliengrenier** [comment]
The tests are failing go: barker@v0.0.0 requires
	testutil@v0.0.0: malformed module path "testutil": missing dot in first path element

**Comment by lbouchard-adgear** [comment]
Yes I noticed ... Working on it.

**Comment by lbouchard-adgear** [comment]
When using `make test`, the dependency was not posing problem, but it was in `barkerd/vcr`. Seems like importing local module in a child module is pausing problem.  To avoid dependency headaches, for now, I move the stuff in `barker/testutil`.  This way it is a package within the same module. Let me know if it is good enough for you.

**Comment by Medulli** [comment]
> When using `make test`, the dependency was not posing problem. But, in `barkerd/vcr`, it seems like importing local module in a child module is pausing problem. To avoid dependency headaches, for now, I move the stuff in `barker/testutil`. This way it is a package within the same module. Let me know if it is good enough for you.

Works for me !

**Pull request merged**